### PR TITLE
Remove references to hermes-engine-darwin

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

As of https://github.com/facebook/hermes/pull/639/ `hermes-engine-darwin` no longer exists, and macOS should use the standard `hermes-engine` that is shared with iOS. This change updates our template to reflect that.

I also updated `local-cli/generator-macos/templates/macOS/Podfile` to more closely match `template/ios/Podfile`. 

## Changelog

[macOS] [Removed] - Remove references to hermes-engine-darwin

## Test Plan

CI Passing should be enough to verify the template still works. `hermes` as of now doesn't work on macOS, so that's still a WIP.

### CODE REVIEW

**Review by HeyImChris — APPROVED** [review]

**Review by kelset — APPROVED** [review]

File: local-cli/generator-macos/templates/macos/Podfile [hunk 1]
```diff
@@ -6,12 +6,7 @@ target 'HelloWorld-macOS' do
   use_native_modules!
   use_react_native!(
     :path => '../node_modules/react-native-macos',
-    
-    # To use Hermes, install the `hermes-engine-darwin` npm package, e.g.:
-    #   $ yarn add 'hermes-engine-darwin@~0.5.3'
-    #
-    # Then enable this option:
-    #   :hermes_enabled => true
+    # :hermes_enabled => true
```
Comment (tido64) [inline local-cli/generator-macos/templates/macos/Podfile:9]: Should we sync to the iOS template with a comment that says what this does? See https://github.com/facebook/react-native/blob/0.66-stable/template/ios/Podfile

File: local-cli/generator-macos/templates/macos/Podfile [hunk 2]
```diff
@@ -3,41 +3,30 @@ require_relative '../node_modules/@react-native-community/cli-platform-ios/nativ
...
   # Enables Flipper.
   #
   # Note that if you have use_frameworks! enabled, Flipper will not work and
-  # you should disable the next line.
-  # use_flipper!()
+  # you should disable these next few lines.
+  use_flipper!
 
+  post_install do |installer|
+    flipper_post_install(installer)
```
Comment (tido64) [inline local-cli/generator-macos/templates/macos/Podfile:27]: `react_native_post_install` should be calling `flipper_post_install`. I think this line is redundant.

File: local-cli/generator-macos/templates/macos/Podfile (hunk 1 above)
Comment (Saadnajmi) [reply local-cli/generator-macos/templates/macos/Podfile:9]: I initially had this as a copy of the iOS template podifle with extra lines for macOS, but that didn't build. I sucessively removed changes till I got to this smallest pod file that does build. I can go back and add the comment again.

File: local-cli/generator-macos/templates/macos/Podfile [hunk 3]
```diff
@@ -6,12 +6,8 @@ target 'HelloWorld-macOS' do
   use_native_modules!
   use_react_native!(
     :path => '../node_modules/react-native-macos',
-    
-    # To use Hermes, install the `hermes-engine-darwin` npm package, e.g.:
-    #   $ yarn add 'hermes-engine-darwin@~0.5.3'
-    #
-    # Then enable this option:
-    #   :hermes_enabled => true
+    # to enable hermes on iOS, change `false` to `true` and then install pods
```
Comment (tido64) [inline local-cli/generator-macos/templates/macos/Podfile:9]:
```suggestion
    # to enable hermes on macOS, change `false` to `true` and then install pods
```

Comment (Saadnajmi) [reply local-cli/generator-macos/templates/macos/Podfile]: Derp, thanks

**Review by tido64 — APPROVED** [review]

**Pull request merged**